src/org/parkerici/image_utils/ij/tiled.clj

```diff
@@ -0,0 +1,93 @@
+(ns org.parkerici.image-utils.ij.tiled
+  (:require [clojure.xml])
+  (:import [loci.plugins BF]
+           [java.io ByteArrayInputStream]
+           [ij IJ ImagePlus ImageStack]
+           [loci.formats.tiff TiffParser]))
+
+(defn parse-xml [s]
+  (clojure.xml/parse (ByteArrayInputStream. (.getBytes s "utf-16"))))
+
+(defn get-tiff-comment [filename]
+  (.getComment (TiffParser. filename)))
+
+(defn read-tiff-metadata [filename]
+  (-> filename get-tiff-comment parse-xml))
+
+(defn get-num-channels
+  "Return the number of channels in an ImagePlus."
+  [^ImagePlus imp]
+  (.getNChannels imp))
+
+(defn get-num-frames
+  "Return the number of frames in an ImagePlus."
+  [^ImagePlus imp]
+  (.getNFrames imp))
+
+(defn get-num-slices
+  "Return the number of slices in an ImagePlus."
+  [^ImagePlus imp]
+  (.getNSlices imp))
+
+(defn get-width
+  "Return the width of an image."
+  [imp]
+  (.getWidth ^ImagePlus imp))
+
+(defn get-height
+  "Return the height of an image."
+  [imp]
+  (.getHeight ^ImagePlus imp))
+
+(defn get-channels
+  [imp]
+  (let [info (.getProperty imp "Info")
+        split-info (clojure.string/split-lines info)
+        name-strings (filterv #(clojure.string/starts-with? % "Name #") split-info)
+        channels (map (fn [x] (last (clojure.string/split x #" = "))) name-strings)
+        channels-underscore (map (fn [x] (clojure.string/replace x #" " "_")) channels)
+        ]
+    channels-underscore))
```

Comment:
You should out some sort of exception/backup in case the TIFF doesn't have an Info field or if the Info field doesn't contain a string with Name.

Reply:
Good call!